File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        results = [0] * len(temperatures)
        numStack = []  # [value, index]

        for i, t in enumerate(temperatures):
            while numStack and t > numStack[-1][0]:
                value, position = numStack.pop()
                results[position] = i - position

            numStack.append([t, i])

        return results

File: Data Structures & Algorithms/daily-temperatures/submission-0.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {

        int[] result = new int[temperatures.length];
        Deque<int[]> stack = new ArrayDeque<>();
        
        for(int i = 0; i < temperatures.length; i++){
            int currentTemp = temperatures[i];

            while (!stack.isEmpty() && stack.peek()[1] < currentTemp) {
                int[] temp = stack.pop();
                int oldIndex = temp[0];

                result[oldIndex] = i - oldIndex;
            }
            stack.push(new int[]{i, currentTemp});
            
        }

        return result;
        
    }
}
